app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/ItineraryShowActivity.kt

```diff
@@ -82,6 +88,14 @@ class ItineraryShowActivity : AppCompatActivity() {
 
         goToMissionInProgressButton = findViewById(R.id.buttonToMissionInProgressActivity)
         goToMissionInProgressButton.isEnabled = droneService.isConnected()
+
+        deleteButton = findViewById(R.id.mission_delete)
+        if (authService.getCurrentSession() != null &&
```

Comment:
I think that using the ? operator for safe call can do the same job in one condition only (authService.getCurrentSession()?.user.uid) and is less cumbersome. But it is a detail

Reply:
Good catch, the code could be refactored to  
 ```deleteButton.visibility = if (authService.getCurrentSession()?.user.uid==ownerUid) View.VISIBLE else View.GONE```